cajita/src/Cajita_Array.hpp

```diff
@@ -204,16 +233,11 @@ class Array
         : _layout( layout )
         , _data( view )
     {
-        if ( (long)view.extent( 0 ) !=
-                 layout->indexSpace( Ghost(), Local() ).extent( 0 ) ||
-             (long)view.extent( 1 ) !=
-                 layout->indexSpace( Ghost(), Local() ).extent( 1 ) ||
-             (long)view.extent( 2 ) !=
-                 layout->indexSpace( Ghost(), Local() ).extent( 2 ) ||
-             (long)view.extent( 3 ) !=
-                 layout->indexSpace( Ghost(), Local() ).extent( 3 ) )
-            throw std::runtime_error(
-                "Layout and view dimensions do not match" );
+        for ( std::size_t d = 0; d < num_space_dim + 1; ++d )
+            if ( (long)view.extent( d ) !=
+                 layout->indexSpace( Ghost(), Local() ).extent( d ) )
```

Comment:
It seems Array is designed over ghost index space but ArrayOp is over own index space. Any reason?

Reply:
The idea here is that point-wise operations can happen on own or own+ghost entities. I wanted to give the user control of this in case they had a large halo region but only needed the owned values update. This is especially true in some PIC variants were we let the halo get pretty large.